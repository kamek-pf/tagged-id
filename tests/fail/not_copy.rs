use tagged_id::Id;

fn main() {
    #[derive(Id)]
    #[tagged_id(String)]
    struct TestStruct();

    let id1: Id<TestStruct> = "oops".into();
    let id2 = id1;

    assert_eq!(id1, id2);
}
