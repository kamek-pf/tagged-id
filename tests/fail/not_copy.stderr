error[E0382]: borrow of moved value: `id1`
  --> tests/fail/not_copy.rs:14:5
   |
11 |     let id1: Id<TestStruct> = "oops".into();
   |         --- move occurs because `id1` has type `Id<TestStruct>`, which does not implement the `Copy` trait
12 |     let id2 = id1;
   |               --- value moved here
13 |
14 |     assert_eq!(id1, id2);
   |     ^^^^^^^^^^^^^^^^^^^^ value borrowed here after move
   |
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)
help: consider cloning the value if the performance cost is acceptable
   |
12 |     let id2 = id1.clone();
   |                  ++++++++
